src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add an order: " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getByUser(User user) {
```

Comment:
```suggestion
    public List<Order> getHistoryByUser(User user) {
```
or
```suggestion
    public List<Order> findByUser(User user) {
```
Because, if you return a list, no orders from that user can be found

Reply:
Sorry? I didn't understand: In "common mistakes" at the link https://mate-academy.github.io/jv-program-common-mistakes/hibernate/add-order/add-order-hw  was specified exactly the "getByUser" name.   http://joxi.ru/DrlPo6xtKk7bPr

Reply:
Approved, I didn't notice that item on the checklist ✅ 
But i still think that a method that may return nothing  shouldn't be named "get"

Reply:
And we'll figure it out